src/js/facility-locator/components/Listbox.jsx

```diff
@@ -0,0 +1,161 @@
+              <span className="flex-center">
+                <span className="legend spacer"></span>
+                {benefitsServices[k] || k}
+              </span>
+            </li>
+          );
+        })
+        }
+      </ul>
+    );
+  }
+
+  renderSelectOptionWithIcon = (facilityType, index) => {
+    const defaultSelected = !this.props.currentQuery.facilityType && !facilityType;
+    const isSelected = this.isSelectedOption((facilityType), 'facilityType');
+    const isHovered = isSelected || defaultSelected;
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true,
+      'is-hovered': isHovered
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <li
+        role="option"
+        tabIndex={this.props.dropdownActive ? '0' : '-1'}
+        aria-selected={isHovered}
+        ref={ elem => { this.props.options[index] = elem; }}
+        id={index > -1 ? (facilityType || 'AllFacilities') : null}
+        className={facilityOptionClasses}
+        onClick={() => this.props.handleFilterSelect(facilityType, 'facility')}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </li>
+    );
+  }
+
+  renderSelectButtonWithIcon = (facilityType) => {
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <div
+        id="facilityDropdown"
+        className={facilityOptionClasses}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </div>
+    );
+  }
+
+  renderSelectOption = (serviceType, isMobile) => {
+    const serviceOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    return (
+      <div className="flex-center">
+        <div
+          id="serviceDropdown"
+          className={serviceOptionClasses}>
+          <span className="flex-center">
+            <span className="legend spacer"></span>
+            {truncate((benefitsServices[serviceType] || serviceType || 'All'), { length: (isMobile ? 38 : 27) })}
+          </span>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { hasIcons, isMobile, isDisabled } = this.props;
+    const { currentQuery: { facilityType, serviceType } } = this.props;
+    return (
+      <div>
+        {hasIcons &&
```

Comment:
I'm having a hard time understanding why the code paths are so different depending on if there are icons or not. It seems like the difference should be limited to how the option and selected value are rendered inside each option. Are there are other reasons for the divergence that I am missing?

Reply:
I'm not sure why the original logic was set up this way, but I will refactor it to make them more similar.